Author: Kai-Chung Yan <seamlikok@gmail.com>
Last-Update: 2015-08-18
--- a/tools/aidl/aidl.cpp
+++ b/tools/aidl/aidl.cpp
@@ -261,7 +261,7 @@
             err |= check_filename(filename, c->package, &c->name);
         }
         else {
-            fprintf(stderr, "aidl: internal error unkown document type %d.\n",
+            fprintf(stderr, "aidl: internal error unknown document type %d.\n",
                         items->item_type);
             return 1;
         }
@@ -1106,13 +1106,13 @@
     }
 
     // write preprocessed file
-    int fd = open( options.outputFileName.c_str(), 
+    int fd = open( options.outputFileName.c_str(),
                    O_RDWR|O_CREAT|O_TRUNC|O_BINARY,
 #ifdef HAVE_MS_C_RUNTIME
                    _S_IREAD|_S_IWRITE);
 #else    
                    S_IRUSR|S_IWUSR|S_IRGRP);
 #endif            
     if (fd == -1) {
         fprintf(stderr, "aidl: could not open file for write: %s\n",
                 options.outputFileName.c_str());
