--- /dev/null
+++ b/libs/androidfw/Makefile
@@ -0,0 +1,55 @@
+#!/bin/make
+
+CXX = g++
+
+# formerly in libutils
+commonUtilsSources:= \
+    Asset.cpp \
+    AssetDir.cpp \
+    AssetManager.cpp \
+    ObbFile.cpp \
+    ResourceTypes.cpp \
+    StreamingZipInflater.cpp
+
+# formerly in libui
+commonUiSources:= \
+    Input.cpp \
+    InputDevice.cpp \
+    Keyboard.cpp \
+    KeyCharacterMap.cpp \
+    KeyLayoutMap.cpp \
+    VelocityControl.cpp \
+    VelocityTracker.cpp \
+    VirtualKeyMap.cpp
+
+commonSources:= \
+	$(commonUtilsSources) \
+	$(commonUiSources)
+
+OBJECTS = $(commonSources:.cpp=.o)
+
+ALL_CXXFLAGS = -O2 -fPIC -I../../include -I/usr/include/android \
+	-include /usr/include/android/arch/linux-x86/AndroidConfig.h
+ALL_LDFLAGS = -fPIC -shared -rdynamic -Wl,-rpath=/usr/lib/android
+ALL_LIBS = -lrt -ldl -lz -lpthread -L/usr/lib/android -lcutils -llog -lutils
+
+CXXFLAGS := $(ALL_CXXFLAGS) $(CXXFLAGS)
+LDFLAGS := $(ALL_LDFLAGS) $(LDFLAGS)
+LIBS := $(ALL_LIBS) $(LIBS)
+
+LIBNAME = libandroidfw
+SONAME = $(LIBNAME).so.0
+SOLIBNAME = $(SONAME).21.0
+
+all: $(LIBNAME).a $(SOLIBNAME)
+
+$(SOLIBNAME): $(OBJECTS)
+	$(CXX) $(LDFLAGS) -Wl,-soname,$(SONAME) -o $(SOLIBNAME) $(OBJECTS) $(LIBS)
+
+$(LIBNAME).a: $(OBJECTS)
+	ar rs $(LIBNAME).a $(OBJECTS)
+
+clean:
+	rm -f $(OBJECTS)
+	rm -f $(LIBNAME).so* $(LIBNAME).a
+
