--- /dev/null
+++ b/tools/aidl/Makefile
@@ -0,0 +1,40 @@
+#!/usr/bin/make
+
+CXX = g++
+
+HOST_OS=linux
+include Android.mk
+
+LEX_SOURCES = $(LOCAL_SRC_FILES:.l=.cpp)
+CXX_SOURCES = $(LEX_SOURCES:.y=.cpp)
+OBJECTS = $(CXX_SOURCES:.cpp=.o)
+
+ALL_CXXFLAGS = -O2 -fPIC $(LOCAL_CFLAGS) \
+	-include /usr/include/android/arch/linux-x86/AndroidConfig.h
+ALL_LDFLAGS = -fPIC -Wl,-rpath=/usr/lib/android
+ALL_LIBS = $(LOCAL_LDLIBS) \
+
+CXXFLAGS := $(ALL_CXXFLAGS) $(CXXFLAGS)
+LDFLAGS := $(ALL_LDFLAGS) $(LDFLAGS)
+LIBS := $(ALL_LIBS) $(LIBS)
+
+# from android-platform-build/core/definitions.mk: transform-l-to-cpp
+%.cpp: %.l
+	echo HERE
+	flex -o$@ $<
+
+# from android-platform-build/core/definitions.mk: transform-y-to-cpp
+%.cpp: %.y
+	bison -o $@ $<
+	touch $(@:$1=.hpp)
+	echo '#ifndef '$(@F:$1=_h) > $(@:$1=.h)
+	echo '#define '$(@F:$1=_h) >> $(@:$1=.h)
+	cat $(@:$1=$(YACC_HEADER_SUFFIX)) >> $(@:$1=.h)
+	echo '#endif' >> $(@:$1=.h)
+
+all: $(OBJECTS)
+	$(CXX) $(LDFLAGS) -o aidl $(OBJECTS) $(LIBS)
+
+clean:
+	rm -f $(OBJECTS)
+	rm -f aidl
